code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -73,6 +73,29 @@
 				if(C.obj_integrity > 1)
 					if(d_type in C.prevent_crits)
 						return TRUE
+
+//This proc returns obj/item/clothing, the armor that has "soaked" the crit. Using it for dismemberment check
+/mob/living/carbon/human/proc/checkcritarmorreference(def_zone, d_type)
+	if(!d_type)
+		return null
+	var/obj/item/clothing/best_armor = null
+	if(isbodypart(def_zone))
+		var/obj/item/bodypart/CBP = def_zone
+		def_zone = CBP.body_zone
+	var/list/body_parts = list(head, wear_mask, wear_wrists, wear_shirt, wear_neck, cloak, wear_armor, wear_pants, backr, backl, gloves, shoes, belt, s_store, glasses, ears, wear_ring)
+	for(var/bp in body_parts)
+		if(!bp)
+			continue
+		if(bp && istype(bp , /obj/item/clothing))
+			var/obj/item/clothing/C = bp
+			if(zone2covered(def_zone, C.body_parts_covered_dynamic))
+				if(C.obj_integrity > 1)
+					if(d_type in C.prevent_crits)
```

Comment:
Since it's bclass that's carried over it's probably better to not label this as d_type in the proc -- that's a whole other variable on weapons / intents